lib/manageiq_performance/commands/profile.rb

```diff
@@ -0,0 +1,268 @@
+require 'manageiq_performance/utils/template_helper'
+
+module ManageIQPerformance
+  module Commands
+    class Profile
+      include TemplateHelper
+
+      attr_reader :output
+
+      DEFAULT_OPTS = {
+        :target_type => :file,
+        :full_require_tree => true,
+        :stackprof_mode => :cpu,
+        :stackprof_interval => 1000
+      }
+
+      def self.help_text
+        "memory/timing profile of a single ruby script or application"
+      end
+
+      def self.run(args)
+        new(args).run
+      end
+
+      def initialize(args)
+        require "pathname"
+
+        @opts         = DEFAULT_OPTS.dup
+        @output       = STDOUT
+        @cmd_root     = Pathname.new Dir.pwd
+        @miqperf_root = Pathname.new File.expand_path("../..", __dir__)
+
+        option_parser.parse!(args)
+        @profile_target  = args.first
+      end
+
+      def run
+        write_tmpfile
+        run_wrapper_script    unless @opts[:debug]
+        debug_wrapper_script  if @opts[:debug]
+      ensure
+        @run_wrapper.close!   if @run_wrapper
+      end
+
+      private
+
+      def write_tmpfile
+        require "erb"
+        require "tempfile"
+
+        @run_wrapper = Tempfile.new("run_wrapper")
+        template_path = "../templates/profile_runner_wrapper.rb.erb"
+        template = File.read File.expand_path(template_path, __dir__)
+
+        @run_wrapper.tap do |wrapper|
+          wrapper.write ERB.new(template, nil, "-").result(binding)
+        end
+      end
+
+      def run_wrapper_script
+        @run_wrapper.close # write the contents of the file to the path
```

Comment:
actually, I was thinking the debug bit could go into the `write_tempfile` bit - and then close in that block of code. so the open and close would all be together. feels better to me for some reason

Reply:
I will try experimenting with this, but the `ensure` in `#run` is the one part I am concerned about and if we `#close` it too early.

Will report back with my tests of this (since I don't have integration tests for this at the moment...), but that will probably be a few days from now since this is currently not on my priority list.